### PR TITLE
Enable role V2 feature

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Enable role V2 feature

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx [hunk 1]
```diff
@@ -16,972 +16,136 @@
...
-
-                if (groupName.length > 1) {
-                    isMatch = re.test(role.display);
-                    if (isMatch) {
-                        filteredRoleList.push(role);
-                        setAssignedRoles(filteredRoleList);
+    const getPlaceholders = () => {
+        if (showEmptyRolesListPlaceholder) {
+            return (
+                // TODO: Need to replace the i18N with the correct one.
```
Comment (dasuni-30) [inline apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx:77]: Shall we resolve the ToDo comments?

File: apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx [hunk 2]
```diff
@@ -195,42 +77,77 @@ export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
...
+     * @param values - Form Values.
+     * @returns Form validation.
+     */
+    const validateForm = async (values: RoleBasicInterface): Promise<RoleBasicInterface> => {
+
+        const errors: RoleBasicInterface = {
+            roleName: undefined
+        };
+
+        if (values.roleName?.toString().trim().length >= 3) {
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx:104]: Any reason to have `length >= 3`?

File: apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx [hunk 3]
```diff
@@ -195,42 +77,77 @@ export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
...
+            try {
+                setRoleNameSearchQuery(`displayName eq ${roleName}`);
+
+                if (!isRolesListLoading || !isRolesListValidating) {
+                    if (rolesList?.totalResults > 0) {
+                        errors.roleName = t("console:manage.features.roles.addRoleWizard.forms."
+                            + "roleBasicDetails.roleName.validations.duplicate",{ type: "Role" });
+                    }
+                } 
+            } catch (error: unknown) {
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx:117]: Shall we add data type?

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 4]
```diff
@@ -88,261 +66,195 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
...
+        if (!isReadOnly && initialSelectedGroupsOptions && selectedGroupsOptions) {
+            setRemovedGroupsOptions(initialSelectedGroupsOptions?.filter((group: GroupsInterface) => {
+                return selectedGroupsOptions?.find(
+                    (selectedGroup: GroupsInterface) => selectedGroup.id === group.id) === undefined;
+            }));
         }
-    };
-
+    }, [ initialSelectedGroupsOptions, selectedGroupsOptions ]);
+
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:148]: Remove whitespace

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:2]: Shall we keep the year as it is?

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 6]
```diff
@@ -88,261 +66,195 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
...
+            setSelectedGroupsOptions(initialSelectedGroupsOptions);
         }
-    }, [ isSelectAssignedAllRolesChecked ]);
 
-    useEffect(() => {
-        if (isSelectUnassignedRolesAllRolesChecked) {
-            setCheckedUnassignedListItems(groupList);
-        } else {
-            setCheckedUnassignedListItems([]);
+        // set groups options when the user is read only
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx:118]:
```suggestion
        // Set groups options when the user is read only
```

File: apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx:77]: This is fixed, will remove the comment.

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx [hunk 7]
```diff
@@ -16,142 +16,429 @@
...
+            }
+        });
+        
+        setInitialAPIResourceIds(uniqueAPIResourceIds);
+        setSelectedPermissions(currentPermissions);
+        setInitialSelectedPermissions(currentPermissions);
+    };
+
+    /**
+     * The following function handles the update of the role permissions.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:220]:
```suggestion
     * Handle the update of the role permissions.
```

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx [hunk 8]
```diff
@@ -16,142 +16,429 @@
...
-                isRole
-                onSubmit={ onPermissionUpdate }
-                roleObject={ roleObject }
-                isSubmitting={ isSubmitting }
-            />
-        </div>
+        <EmphasizedSegment padded="very">
+            <Grid xs={ 8 }>
+                <Heading as="h4">
+                    Assigned Permissions
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:335]: Shall we add i18n?

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx [hunk 9]
```diff
@@ -16,142 +16,429 @@
...
+                </Heading>
+                <Heading as="h6" color="grey" subHeading className="mb-5">
+                    Manage assigned permissions in the role.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:338]: Shall we add i18n?

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx [hunk 10]
```diff
@@ -16,142 +16,429 @@
...
+                    <DocumentationLink
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:339]: Do we need the learn more link for IS?

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx [hunk 11]
```diff
@@ -16,142 +16,429 @@
...
+                                        { t("console:manage.features.roles.addRoleWizard.forms.rolePermission." +
+                                            "permissions.label") }
+                                    </label>
+                                    <EmphasizedSegment
+                                        className="mt-2"
+                                        data-componentid={ componentId }
+                                        basic
+                                        loading={ 
+                                            selectedAPIResourceId &&
+                                            (isSelectedAPIResourceFetchRequestLoading
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:389]: Let's fix indentation

File: apps/console/src/features/roles/pages/role.tsx [hunk 12]
```diff
@@ -295,43 +149,47 @@ const RolesPage = (): ReactElement => {
...
-        setSearchQuery(null);
-        getRoles();
+        setFilterBy(null);
     };
 
+    const handleCreateRole = () => {
+        history.push(AppConstants.getPaths().get("ROLE_CREATE"));
+    };
+        
+
```
Comment (DonOmalVindula) [inline apps/console/src/features/roles/pages/role.tsx:174]: Remove whitespace

**Review by DonOmalVindula — APPROVED** [review]

File: apps/console/src/features/roles/pages/role.tsx (hunk 12 above)
Comment (Achintha444) [reply apps/console/src/features/roles/pages/role.tsx:173]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 4 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx (hunk 11 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx:383]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx (hunk 10 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx (hunk 9 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx (hunk 8 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx (hunk 7 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx [hunk 13]
```diff
@@ -352,426 +264,279 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
...
+                                                    displayName={ RoleManagementUtils.getGroupDisplayName(option) }
+                                                    userstore={ RoleManagementUtils.getUserStore(option.displayName) }
+                                                    renderOptionProps={ props }
+                                                />
+                                            ) }
+                                        />
+                                    ) : null
+                            } 
+
+                            { /* Update Button */ }
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 6 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx (hunk 5 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx (hunk 3 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

File: apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7

**Review by dasuni-30 — APPROVED** [review]

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**